server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -1212,14 +1216,21 @@ public boolean resourceStateTransitTo(final Host host, final ResourceState.Event
 
     private boolean doMaintain(final long hostId) {
         final HostVO host = _hostDao.findById(hostId);
+        s_logger.info("Maintenance: attempting maintenance of host " + host.getUuid());
+        ResourceState hostState = host.getResourceState();
+        if (hostState == ResourceState.PrepareForMaintenance || hostState == ResourceState.ErrorInPrepareForMaintenance ||
```

Comment:
Same as before,  using `Array/list::contains()` or equivalent. If this is frequenty used code, refactor as a separate internal method and re-use wherever needed.